flume-ng-channels/flume-file-channel/src/main/java/org/apache/flume/channel/file/instrumentation/FileChannelCounter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.flume.channel.file.instrumentation;
+
+import org.apache.flume.channel.file.FileChannel;
+import org.apache.flume.instrumentation.ChannelCounter;
+
+public class FileChannelCounter extends ChannelCounter implements FileChannelCounterMBean {
+
+  private final FileChannel fileChannel;
+
+  public FileChannelCounter(FileChannel fileChannel) {
+    super(fileChannel.getName());
+    this.fileChannel = fileChannel;
+  }
+
+  public boolean isOpen() {
```

Comment:
Thank you for your review. I wasn't sure which way to go, because I didn't want to put the responsibility of updating the flag to the FileChannel. (Which shouldn't be a problem, to be honest, as all the other counters work like this).
Thanks for the good example, I agree with you and I'll update my patch according to your comment.